registry/client/repository.go

```diff
@@ -99,10 +98,6 @@ func (r *registry) Repositories(ctx context.Context, entries []string, last stri
 
 // NewRepository creates a new Repository for the given repository name and base URL.
 func NewRepository(ctx context.Context, name, baseURL string, transport http.RoundTripper) (distribution.Repository, error) {
-	if _, err := reference.ParseNamed(name); err != nil {
-		return nil, err
-	}
-
```

Comment:
Opened https://github.com/docker/distribution/pull/1270. We can wait for this PR to get merged first to avoid a rebase nightmare.